examples/gno.land/p/demo/btree/btree.gno

```diff
@@ -0,0 +1,1114 @@
+func (t *BTree) Len() int {
+	return t.length
+}
+
+// Clear removes all elements from the B-tree.
+//
+// Parameters:
+// - addNodesToFreelist: 
+//     - If true, the tree's nodes are added to the freelist during the clearing process, 
+//       up to the freelist's capacity.
+//     - If false, the root node is simply dereferenced, allowing Go's garbage collector 
+//       to reclaim the memory.
+//
+// Benefits:
+// - **Performance:** 
+//     - Significantly faster than deleting each element individually, as it avoids the overhead 
+//       of searching and updating the tree structure for each deletion.
+//     - More efficient than creating a new tree, since it reuses existing nodes by adding them 
+//       to the freelist instead of discarding them to the garbage collector.
+//
+// Time Complexity:
+// - **O(1):** 
+//     - When `addNodesToFreelist` is false.
+//     - When `addNodesToFreelist` is true but the freelist is already full.
+// - **O(freelist size):** 
+//     - When adding nodes to the freelist up to its capacity.
+// - **O(tree size):** 
+//     - When iterating through all nodes to add to the freelist, but none can be added due to 
+//       ownership by another tree.
+
+func (tree *BTree) Clear(addNodesToFreelist bool) {
+    if tree.root != nil && addNodesToFreelist {
+        tree.root.reset(tree.cowCtx)
+    }
+    tree.root = nil
+    tree.length = 0
+}
+
+// reset adds all nodes in the current subtree to the freelist.
+//
+// The function operates recursively:
+// - It first attempts to reset all child nodes.
+// - If the freelist becomes full at any point, the process stops immediately.
+//
+// Parameters:
+// - copyOnWriteCtx: The copy-on-write context managing the freelist.
+//
+// Returns:
+// - true: Indicates that the parent node should continue attempting to reset its nodes.
+// - false: Indicates that the freelist is full and no further nodes should be added.
+//
+// Usage:
+// This method is called during the `Clear` operation of the B-tree to efficiently reuse
+// nodes by adding them to the freelist, thereby avoiding unnecessary allocations and reducing
+// garbage collection overhead.
+func (currentNode *node) reset(copyOnWriteCtx *copyOnWriteContext) bool {
+    // Iterate through each child node and attempt to reset it.
+    for _, childNode := range currentNode.children {
+        // If any child reset operation signals that the freelist is full, stop the process.
+        if !childNode.reset(copyOnWriteCtx) {
+            return false
+        }
+    }
+
+    // Attempt to add the current node to the freelist.
+    // If the freelist is full after this operation, indicate to the parent to stop.
+    freelistStatus := copyOnWriteCtx.freeNode(currentNode)
+    return freelistStatus != ftFreelistFull
+}
+
+// Int implements the Record interface for integers.
+//
+// TODO: Is there any advantage to implementing this interface for all of the standard
+// orderable types?
```

Comment:
This can make sense as a subpackage, like `p/demo/btree/cmp`

Reply:
Good point. I removed the comments, and can implement this subpackage later.

Reply:
The comments were removed.